sormas-backend/src/main/java/de/symeda/sormas/backend/action/ActionService.java

```diff
@@ -0,0 +1,220 @@
+			Predicate userFilter = createUserFilter(cb, cq, from);
+			filter = AbstractAdoService.and(cb, filter, userFilter);
+		}
+
+		cq.where(filter);
+		cq.select(from.get(Action.UUID));
+
+		return em.createQuery(cq).getResultList();
+	}
+
+	public List<Action> getAllByEvent(Event event) {
+
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<Action> cq = cb.createQuery(getElementClass());
+		Root<Action> from = cq.from(getElementClass());
+
+		Predicate filter = cb.equal(from.get(EventParticipant.EVENT), event);
+		cq.where(filter);
+		cq.orderBy(cb.desc(from.get(EventParticipant.CREATION_DATE)));
+
+		return em.createQuery(cq).getResultList();
+	}
+
+	/**
+	 * @see /sormas-backend/doc/UserDataAccess.md
+	 */
+	@SuppressWarnings("rawtypes")
+	@Override
+	public Predicate createUserFilter(CriteriaBuilder cb, CriteriaQuery cq, From<Action, Action> actionPath) {
+
+		// National users can access all actions in the system
+		User currentUser = getCurrentUser();
+		if (currentUser.hasAnyUserRole(UserRole.NATIONAL_USER, UserRole.NATIONAL_CLINICIAN, UserRole.NATIONAL_OBSERVER, UserRole.REST_USER)) {
+			return null;
+		}
+
+		// whoever created the action is allowed to access it
+		Predicate filter = cb.equal(actionPath.join(Action.CREATOR_USER, JoinType.LEFT), currentUser);
+
+		Predicate eventFilter = eventService.createUserFilter(cb, cq, actionPath.join(Action.EVENT, JoinType.LEFT));
+		if (eventFilter != null) {
+			filter = cb.or(filter, eventFilter);
+		}
+
+		return filter;
+	}
+
+	private Predicate buildActiveActionsFilter(CriteriaBuilder cb, Root<Action> from) {
+
+		Join<Action, Event> event = from.join(Action.EVENT, JoinType.LEFT);
+
+		Predicate filter = cb.and(
+				cb.equal(from.get(Action.ACTION_CONTEXT), ActionContext.EVENT),
+				cb.or(cb.equal(event.get(Event.ARCHIVED), false), cb.isNull(event.get(Event.ARCHIVED))));
+
+		return filter;
+	}
+
+	/**
+	 * Computes stats for action matching an actionCriteria.
+	 */
+	public List<ActionStatEntry> statsByEvent(ActionCriteria actionCriteria) {
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<ActionStatEntry> cq = cb.createQuery(ActionStatEntry.class);
+		Root<Action> action = cq.from(getElementClass());
+		cq.multiselect(cb.countDistinct(action.get(Action.ID)).alias("count"), action.get(Action.ACTION_STATUS));
+
+		Predicate filter = null;
+		if (actionCriteria == null || !actionCriteria.hasContextCriteria()) {
+			filter = createUserFilter(cb, cq, action);
+		}
+
+		if (actionCriteria != null) {
+			Predicate criteriaFilter = buildCriteriaFilter(actionCriteria, cb, action);
+			filter = AbstractAdoService.and(cb, filter, criteriaFilter);
+		}
+
+		if (filter != null) {
+			cq.where(filter);
+		}
+		cq.groupBy(action.get(Action.ACTION_STATUS));
+		cq.orderBy(cb.desc(action.get(Action.ACTION_STATUS)));
+		return em.createQuery(cq).getResultList();
+	}
+
+	public List<Action> getList(ActionCriteria actionCriteria, Integer first, Integer max) {
```

Comment:
Rename to getActionList for clarification

Reply:
OK